miner/src/worker/dummy.rs

```diff
@@ -94,10 +94,29 @@ impl Dummy {
         }
     }
 
-    fn solve(&self, pow_hash: Byte32, work: Work, nonce: u128) {
-        thread::sleep(self.delay.duration());
-        if let Err(err) = self.nonce_tx.send((pow_hash, work, nonce)) {
-            error!("nonce_tx send error {:?}", err);
+    fn solve(&mut self, mut pow_hash: Byte32, mut work: Work, nonce: u128) {
+        let instant = Instant::now();
+        let delay = self.delay.duration();
+        loop {
+            thread::sleep(Duration::from_millis(10));
+            if instant.elapsed() > delay {
+                if let Err(err) = self.nonce_tx.send((pow_hash, work, nonce)) {
+                    error!("nonce_tx send error {:?}", err);
+                }
+                return;
+            }
+            // if there is new work and pow_hash changed, start working on the new one
+            if let Ok(WorkerMessage::NewWork {
+                pow_hash: new_pow_hash,
+                work: new_work,
+                ..
+            }) = self.worker_rx.try_recv()
+            {
+                if new_pow_hash != pow_hash {
+                    pow_hash = new_pow_hash;
+                    work = new_work;
+                }
+            }
```

Comment:
How about this:
```rust
    fn solve(&self, pow_hash: Byte32, work: Work, nonce: u128) {
        select! {
            recv(self.worker_rx) -> msg => {
                if let Ok(WorkerMessage::NewWork {
                        pow_hash: new_pow_hash,
                        work: new_work,
                        ..
                }) = msg {
                    if new_pow_hash != pow_hash {
                        pow_hash = new_pow_hash;
                        work = new_work;
                    }
                }
            },
            default(self.delay.duration()) => {
                if let Err(err) = self.nonce_tx.send((pow_hash, work, nonce)) {
                    error!("nonce_tx send error {:?}", err);
                }
                return;

            }
        }
    }
```

Reply:
The current develop branch use a simple thread sleep, and I wanted to keep the code as unchanged as possible, rather than introduce a tokio runtime.

Reply:
> rather than introduce a tokio runtime.

OK, but `self.nonce_tx`'s type is `crossbeam_channel::Sender`. We won't need tokio runtime here.

Reply:
I see, I thought it's tokio's select macro, let me change it and test later, thanks.

Reply:
after test, I found the code generated by `select` macro is different, the `pow_hash` and `work` vars are not read again, I will keep current code.